src/Functions/keyvaluepair/src/impl/state/State.h

```diff
@@ -0,0 +1,34 @@
+#pragma once
+
+#include <cstddef>
+
+namespace DB
+{
+
+enum State
+{
+    WAITING_KEY,
+    READING_KEY,
+    READING_ENCLOSED_KEY,
+    READING_KV_DELIMITER,
+    WAITING_VALUE,
+    READING_VALUE,
+    READING_ENCLOSED_VALUE,
+    READING_EMPTY_VALUE,
+    FLUSH_PAIR,
+    END
+};
+
+struct NextState
+{
+    std::size_t pos;
+    State state;
+};
+
+struct NextStateWithElement
+{
+    NextState state;
+    std::string_view element;
```

Comment:
What is element?

Reply:
This struct is not even used anymore, forgot to delete it. In any case, an element is either a key or a value.